parfive/tests/test_downloader.py

```diff
@@ -343,3 +349,78 @@ def test_ftp_http(tmpdir, httpserver):
     f = dl.download()
     assert len(f) == 2
     assert len(f.errors) == 4
+
+
+def test_default_user_agent(event_loop, httpserver, tmpdir):
+    tmpdir = str(tmpdir)
+    httpserver.serve_content('SIMPLE  = T',
+                             headers={'Content-Disposition': "attachment; filename=testfile.fits"})
+
+    dl = Downloader(loop=event_loop)
+    dl.enqueue_file(httpserver.url, path=Path(tmpdir), max_splits=None)
+
+    assert dl.queued_downloads == 1
+
+    dl.download()
+
+    assert 'User-Agent' in httpserver.requests[0].headers
+    assert httpserver.requests[0].headers['User-Agent'] == f"parfive/{parfive.__version__} aiohttp/{aiohttp.__version__} python/{sys.version[:5]}"
+
+
+def test_custom_user_agent(event_loop, httpserver, tmpdir):
+    tmpdir = str(tmpdir)
+    httpserver.serve_content('SIMPLE  = T',
+                             headers={'Content-Disposition': "attachment; filename=testfile.fits"})
+
+    dl = Downloader(loop=event_loop, headers={'User-Agent': 'test value 299792458'})
+    dl.enqueue_file(httpserver.url, path=Path(tmpdir), max_splits=None)
+
+    assert dl.queued_downloads == 1
+
+    dl.download()
+
+    assert 'User-Agent' in httpserver.requests[0].headers
+    assert httpserver.requests[0].headers['User-Agent'] == "test value 299792458"
+
+
+@pytest.fixture
+def create_session(event_loop):
+    session = None
+
+    async def maker(*args, **kwargs):
+        nonlocal session
+        session = aiohttp.ClientSession(*args, **kwargs)
+        return session
+    yield maker
+    if session is not None:
+        event_loop.run_until_complete(session.close())
+
+
+@pytest.fixture
+def session(create_session, event_loop):
+    return event_loop.run_until_complete(create_session())
+
+def test_proxy_passed_as_kwargs_to_get(session, event_loop, tmpdir):
+    k = mock.patch.dict(os.environ,{'PROXY': "proxy_url",'PROXY_AUTH': "proxy_auth"})
```

Comment:
I would like to use `pytest-mock` and the `mocker` fixture here, as it automatically does cleanup etc.

Reply:
Does this work?